packages/workflow/src/types.ts

```diff
@@ -75,39 +77,40 @@ export interface PublicEngineAction {
      * edges if an action contains `step.waitForEvent`.
      */
     edges?: Array<PublicEngineEdge>;
-  }
-};
+  };
+}
 
 export type PublicEngineEdge = Omit<WorkflowEdge, "from" | "to">;
 
 /**
  * EngineAction represents a reusable action, or step, within a workflow.  It defines the
  * kind, the handler to run, the types for the action, and optionally custom UI for managing
  * the action's configuration within the workflow editor.
- * 
+ *
  * Note that this is the type representing an action in the *backend engine*.
  *
  */
-export interface EngineAction extends PublicEngineAction {
+export interface EngineAction<TClient extends Inngest = Inngest>
+  extends PublicEngineAction {
   /**
    * The handler is the function which runs the action.  This may comprise of
-   * many individual inngest steps. 
+   * many individual inngest steps.
    */
-  handler: ActionHandler;
-};
+  handler: ActionHandler<ReturnType<typeof createStepTools<TClient>>>;
```

Comment:
@jpwilliams I'd love your input on that 👀

Reply:
Use `GetStepTools`, as that's our publicly-supported version: https://www.inngest.com/docs/typescript#get-step-tools

```ts
import { type GetStepTools } from "inngest";
import { inngest } from "@/inngest";

type StepTools = GetStepTools<typeof inngest>;
type StepToolsWithTrigger = GetStepTools<typeof inngest, "app/user.created">;
```